x-pack/plugins/security_solution/public/detection_engine/rule_management/logic/bulk_actions/show_bulk_success_toast.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { BulkActionSummary } from '..';
+import type { UseAppToasts } from '../../../../common/hooks/use_app_toasts';
+import type { BulkAction } from '../../../../../common/detection_engine/rule_management/api/rules/bulk_actions/request_schema';
+import { explainBulkSuccess, summarizeBulkSuccess } from './translations';
+
+export function showBulkSuccessToast(
+  toasts: UseAppToasts,
```

Comment:
It would probably make sense to transform this function into a hook, so `toasts` will become a detail of internal implementation rather than passed from outside. It could be used then as follows:

```tsx
// No knowledge about the `toasts` service on the call side
const showBulkSuccessToast = useShowBulkSuccessToast()
showBulkSuccessToast(action, summary)
```

Reply:
It's impossible to use hooks inside usual functions. `downloadExportedRules` uses `showBulkSuccessToast` as well.

Reply:
Okay, I think `downloadExportedRules` should not know anything about toasts and provide generic `onError` and `onSuccess` callbacks instead. But let's leave this improvement for another time.